hs_access_control/models/group.py

```diff
@@ -113,24 +178,97 @@ def members(self):
                                    u2ugp__group=self.group,
                                    u2ugp__privilege__lte=PrivilegeCodes.VIEW)
 
+    @property
+    def viewers(self):
+        """ a viewer is not necessarily a member, due to community influence """
+        return User.objects.filter(
+                Q(is_active=True) &
+                (Q(u2ugp__group__gaccess__active=True,
+                   u2ugp__group=self.group) |
+                 Q(u2ugp__group__gaccess__active=True,
+                   u2ugp__group__g2gcp__community__c2gcp__group__gaccess__active=True,
+                   u2ugp__group__g2gcp__community__c2gcp__group=self.group))).distinct()
+
+    def communities(self):
+        """
+        Return list of communities of which this group is a member.
+
+        :return: list of communities
+
+        """
+        return Community.objects.filter(c2gcp__group=self.group)
+
+    @property
+    def __view_resources_of_group(self):
+        """
+        resources viewable according to group privileges
+
+        Used in queries of BaseResource
+        """
+        return Q(r2grp__group=self.group)
+
+    @property
+    def __edit_resources_of_group(self):
+        """
+        resources editable according to group privileges
+
+        Used in queries of BaseResource
+        """
+        return Q(r2grp__group=self.group,
+                 raccess__immutable=False,
+                 r2grp__privilege__lte=PrivilegeCodes.CHANGE)
+
+    @property
+    def __view_resources_of_community(self):
+        """
+        Subquery Q expression for viewable resources according to community memberships
+
+        Used in BaseResource queries only
+        """
+        return Q(r2grp__group__gaccess__active=True,
+                 r2grp__group__g2gcp__allow_view=True,
+                 r2grp__group__g2gcp__community__c2gcp__group=self.group)
+
+    @property
+    def __edit_resources_of_community(self):
+        """
+        Subquery Q expression for editable resources according to community memberships
+
+        Used in BaseResource queries only.
+        """
+        return Q(raccess__immutable=False,
+                 r2grp__group__gaccess__active=True,
+                 r2grp__privilege=PrivilegeCodes.CHANGE,
```

Comment:
Should this be `r2grp__privilege__lte=PrivilegeCodes.CHANGE`?

Reply:
Well, I know that it can't be < CHANGE, so I didn't allow that. Groups can't own anything.